File: solutions/c/day-1.c
int findClosestNumber(int* nums, int numsSize) {
    int ans = 9999999;
    int is_neg = 0;

    for (int i = 0; i < numsSize; i++) {
        if (nums[i] < 0) {
            int tmp = nums[i] * -1;
            if (tmp < ans) {
                ans = tmp;
                is_neg = 1;
            }
        } else {
            int tmp = nums[i];
            if (tmp < ans) {
                ans = tmp;
                is_neg = 0;
            } else if (tmp == ans) {
                is_neg = 0;
            }
        }
    }

    if (is_neg) {
        ans = ans * -1;
    }

    return ans;
}

File: solutions/rust/day-1.rs
impl Solution {
    pub fn find_closest_number(nums: Vec<i32>) -> i32 {
        let mut ans = 999999;
        let mut is_neg: bool = false;
        for num in nums.iter() {
            if (*num < 0) {
                let tmp = num * -1;
                if (tmp < ans) {
                    ans = tmp;
                    is_neg = true;
                }
            } else {
                if (*num < ans) {
                    ans = *num;
                    is_neg = false
                } else if (*num == ans) {
                    is_neg = false
                }
            }
        }

        if (is_neg) {
            ans = ans * -1;
        }

        return ans;
    }
}
